lib/app/modules/smart/dynamic_registration_sequence.rb

```diff
@@ -31,8 +31,8 @@ class DynamicRegistrationSequence < SequenceBase
 
       show_uris
 
-      requires :oauth_register_endpoint, :client_name, :initiate_login_uri, :redirect_uris, :scopes, :confidential_client,:initiate_login_uri, :redirect_uris, :dynamic_registration_token
```

Comment:
This is incorrect -- for Dynamic Registration we need to indicate whether we are registering a confidential or public client, but we do not "input" the client secret.  Instead, we receive the client secret from the dynamic registration server. Could you revert this and let me know if your solution still works?

Reply:
Reverted change and the solution still works.